### PR TITLE
Add types for Structured Metadata functions

### PR DESCRIPTION
**Pull request by strausr**

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -774,6 +807,40 @@ declare module 'cloudinary' {
...
+
+            /****************************** Structured Metadata API V2 Methods *************************************/
+
+            function add_metadata_field(field: MetadataField, options?: AdminApiOptions, callback?: ResponseCallback): Promise<MetadataField>;
+
+            function add_metadata_field(field: MetadataField, callback?: ResponseCallback): Promise<MetadataField>;
+
+            function list_metadata_fields(callback?: ResponseCallback, options?: AdminApiOptions): Promise<MetadataFields>;
+
+            function list_metadata_fields(options?: AdminApiOptions): Promise<MetadataFields>;
```
Comment (patrick-tolosa) [inline types/index.d.ts]: I think there's a difference between the response of MetadataFields and the argument of MetadataFields

The argument MetadataFields can have missing parts, can the response also have missing parts?
For example, if MetadataField.label is optional, when you fetch it from the server, is will that key be missing?

File: types/index.d.ts [hunk 2]
```diff
@@ -555,6 +555,39 @@ declare module 'cloudinary' {
...
+        external_id?: string;
+        type?: string;
+        label?: string;
+        mandatory?: boolean;
+        default_value?: number;
+        validation?: object;
+        datasource?: object;
+    }
+
+    export interface MetadataFields {
```
[the remark this answers is not in the document]
Comment (strausr) [reply types/index.d.ts]: yup. changing

File: types/index.d.ts (hunk 1 above)
Comment (strausr) [reply types/index.d.ts]: Created two separate interfaces. one for the response and one for the options.

File: types/index.d.ts [hunk 3]
```diff
@@ -555,6 +555,43 @@ declare module 'cloudinary' {
...
+        datasource: object;
+
+        [futureKey: string]: any;
+    }
+
+    export interface MetadataFields {
+        metadata_fields: MetadataFieldApiResponse[]
+    }
+
+    export interface datasourceChange {
```
Comment (patrick-tolosa) [inline types/index.d.ts]: Capital D!

File: types/index.d.ts [hunk 4]
```diff
@@ -555,6 +555,43 @@ declare module 'cloudinary' {
...
+        label: string;
+        mandatory: boolean;
+        default_value: number;
+        validation: object;
+        datasource: object;
+
+        [futureKey: string]: any;
+    }
+
+    export interface MetadataFields {
```
Comment (patrick-tolosa) [inline types/index.d.ts]: MetadataFieldsApiResponse maybe?

File: types/index.d.ts [hunk 5]
```diff
@@ -80,10 +79,6 @@ declare module 'cloudinary' {
         | "assist_colorblind"
         | "recolor"
         | "tint"
-        | "contrast"
```
Comment (asisayag2) [inline types/index.d.ts]: @strausr. why were those removed?

File: types/cloudinary_ts_spec.ts [hunk 6]
```diff
@@ -568,6 +568,42 @@ cloudinary.v2.api.usage(function (error, result) {
 // $ExpectType Promise<any>
 cloudinary.v2.api.usage({public_id: 'demo'});
 
+cloudinary.v2.api.add_metadata_field({
```
Comment (asisayag2) [inline types/cloudinary_ts_spec.ts]: @strausr , isn't the $ExpectType comment important here?

Comment (strausr) [reply types/cloudinary_ts_spec.ts]: No. We can't check the type of the promise bc call_api is not really called and will always return nothing. Its enough to verify the promise with .then() and a sanity check for us that the functions are accepting the proper fields
